### ISSUE TITLE
operator-name in csv-config.yaml is ignored

### ISSUE THREAD

**Issue by aliok** [operator-framework/operator-sdk#2266]
## Bug Report

**What did you do?**
I ran this command:
```
operator-sdk olm-catalog gen-csv            \
  --csv-config    hack/csv-config.yaml      \
  --from-version  0.10.1                    \
  --csv-version   0.10.2                    \
  [...]
```

with `hack/csv-config.yaml`:
```
operator-name: knative-eventing-operator
[...]
```
Note that, operator's directory name does not match the operator name in `csv-config.yaml`.
Operator directory is called `eventing-operator`.

**What did you expect to see?**

* A new directory created as `deploy/olm-catalog/knative-eventing-operator/0.10.2`.
* CSV file with `name` field as `knative-eventing-operator.v0.10.2`

**What did you see instead? Under which circumstances?**

* A new directory created as `deploy/olm-catalog/eventing-operator/0.10.2`.
* CSV file with `name` field as `eventing-operator.v0.10.2`

**Environment**
* operator-sdk version:

operator-sdk version: "v0.12.0", commit: "2445fcda834ca4b7cf0d6c38fba6317fb219b469", go version: "go1.13.3 linux/amd64"

* go version:

go version go1.13.3 linux/amd64

* Are you writing your operator in ansible, helm, or go?

This operator is not an Operator-SDK based operator. However, I think the `gen-csv` feature can still be used if some improvements are done.

**Possible Solution**

Pass `--operator-name` param.

**Additional context**

Operator name is read as the directory name here: https://github.com/operator-framework/operator-sdk/blob/424a61d56000e6e3d91d352faa1bd4f7c814661f/cmd/operator-sdk/olmcatalog/gen-csv.go#L94

I guess that is carried over!

**Comment by camilamacedo86** [comment]
hi @aliok,

Thank you for raising it. 

 c/c @estroz @joelanford

**Comment by estroz** [comment]
@aliok thanks for filing an issue. This is fixed in #2249, which removes config file support in favor of `--exclude=[paths]`. You must set `--operator-name` explicitly if you want a different operator name for your CSV.

**Comment by estroz** [comment]
@aliok see #2297 for the immediate fix.
